navigator_api/api/main.py

```diff
@@ -93,79 +96,55 @@ def workflow_state(dataset_id):
-                "title": "Update Spectrum",
-                "completed": False,
-                "progress": 0
-            }
-        ],
-        "taskBreadcrumps": [
-            "aaabbbbcc",
-            "dddeeefff",
-            "asdfasdfa",
-            "aaabbbbcc",
-            "dddeeefff",
-            "asdfasdfa",
-            "aaabbbbcc",
-            "dddeeefff",
-            "asdfasdfa"
-        ],
+        "progress": engine_decision["progress"]["progress"],
+        "milestones": engine_decision["progress"]["milestones"],
+        "milestoneListFullyResolved": engine_decision["progress"]["milestoneListFullyResolved"],
+        "taskBreadcrumps": engine_decision["actions"],
         "currentTask": {
-            "id": "asdfasdfa",
-            "skipped": False,
-            "details": {
-                "milestoneId": "zzz",
-                "title": "Populate ART template",
-                "displayHtml": "<p><strong>Lorem Ipsum</strong> is simply dummy <br /> "
-                               "text of the printing and typesetting industry.</p>",
-                "skippable": True,
-                "helpUrls": [
-                    {"label": "Naomi help docs", "url": "http://example"},
-                    {"label": "Spectrum documentation", "url": "http://example"}
-                ]
-            }
+            "id": engine_decision["decision"]["id"],
+            "skipped": is_task_skipped(dataset_id, engine_decision["decision"]["id"]),
+            "details": _mock_task_details(engine_decision["decision"]["content"])
         }
     })
 
 
+def is_task_skipped(dataset_id, task_id):
+    workflow = model.get_workflow(dataset_id, current_user.id)
+    return task_id in workflow.skipped_tasks
+
+
 @main_blueprint.route('/workflows/<dataset_id>/tasks/<task_id>')
 @login_required
 def workflow_task_details(dataset_id, task_id):
+    try:
+        task_details = engine_client.get_action(task_id)
+    except Exception:
+        log.exception(f"Failed to get task details {task_id}", exc_info=True)
+        return error.not_found(f"Failed to get task details {task_id}")
     return jsonify({
         "id": f"{task_id}",
-        "skipped": False,
-        "details": {
-            "milestoneId": "zzz",
-            "title": "Populate ART template",
-            "displayHtml": "<p><strong>Lorem Ipsum</strong> is simply dummy <br /> "
-                           "text of the printing and typesetting industry.</p>",
-            "skippable": True,
-            "actionUrl": "https://dev.adr.fjelltopp.org/datasets",
-            "helpUrls": [
-                {"label": "Naomi help docs", "url": "http://example"},
-                {"label": "Spectrum documentation", "url": "http://example"}
-            ]
-        }
+        "skipped": is_task_skipped(dataset_id, task_id),
+        "details": _mock_task_details(task_details)
     })
 
 
+def _mock_task_details(task_details):
+    mock = {
+        "milestoneId": "6",
```

Comment:
We should talk about why you are adding `milestoneId` in here.

Reply:
the UI is displaying the current milestone of a browsed task.